evaluator/evaluator.go

```diff
@@ -364,6 +362,19 @@ func evalReturnStatement(rs *ast.ReturnStatement, env *object.Environment) objec
 	return &object.Return{Value: val}
 }
 
+func evalConstStatement(cs *ast.ConstStatement, env *object.Environment) object.Object {
+	if _, ok := env.ShallowGet(cs.Name.Value); ok {
+		return newError("redeclared constant: %q in one block", cs.Name.Value)
+	}
+
+	val := Eval(cs.Value, env)
+	if isError(val) {
+		return val
+	}
+
+	return env.Set(cs.Name.Value, val)
+}
+
```

Comment:
that's old `VarStatement` handling with additional checking if constant's identifier was already declared in current block-scope.